include/graphblas/hyperdags/blas3.hpp

```diff
@@ -40,11 +40,10 @@ namespace grb {
 	template<
 		Descriptor descr = descriptors::no_operation,
 		typename OutputType, typename InputType1, typename InputType2,
-		typename RIT, typename CIT, typename NIT,
 		class MulMonoid
 	>
 	RC eWiseApply(
-		Matrix< OutputType, hyperdags, RIT, CIT, NIT > &C,
+		Matrix< OutputType, hyperdags > &C,
 		const Matrix< InputType1, hyperdags > &A,
```

Comment:
These look like programming errors -- is it the case there was/is no unit test for this eWiseApply variant? (Is there an issue for that?)

Reply:
I can extend the _eWiseApply_matrix_ test for the Monoid variant if you want @anyzelman

Reply:
If these are not programming errors (i.e., the code compiles when called, which I surmise in one case does work), then removing these template args means that they are assumed to revert back to their defaults, correct? In that case, it would mean passing a matrix instantiated with non-default RIT, CIT, NIT would not compile, isn't that correct?

But in such a case the removed code also does not seem to be correct, because it only applies it on C, and not on e.g. A. And potentially it is also incorrect elsewhere in the HyperDAGs backend. The thing that we then apparently do not test for, is the use of grb::Matrix with non-standard row, column, or nonzero indices.

If these all look like correct statements to you, then the best thing to do is probably to open up an issue about reviewing the Matrix signatures throughout the code base, and make sure that when they're public facing, it's templated in RIT, CIT, and NIT everywhere and for all arguments separately, and to happy path test them on all related primitives? (It's a lot of work, and imo not very high priority)

Reply:
And a separate issue for eWiseApply_matrix test for the Monoid variant, that would be higher priority indeed=)

Reply:
> If these are not programming errors (i.e., the code compiles when called, which I surmise in one case does work), then removing these template args means that they are assumed to revert back to their defaults, correct? In that case, it would mean passing a matrix instantiated with non-default RIT, CIT, NIT would not compile, isn't that correct?

You are right, I misunderstood. 
I assumed that the template parameters would be deduced which I realise now does not make sense.
Will fix it.

Reply:
Fixed [here](https://github.com/Algebraic-Programming/ALP/pull/184/commits/6f15aa45dbcaf4f4d2f29e1ac677f2b200cd6649)

Reply:
Thanks! I fixed similar errors throughout the code base under this MR just now, see latest commit. Will test the changes via CI as well as manually.

(It still does not include the tests)

Reply:
Ah perfect! You've been faster than me

Reply:
> And a separate issue for eWiseApply_matrix test for the Monoid variant, that would be higher priority indeed=)

I just realised that I caught this a few weeks ago: https://github.com/Algebraic-Programming/ALP/issues/203

Reply:
Oh great! Should I help out with that one?

Reply:
Meanwhile flagging this thread as resolved (tests will be added as part of #203 and #205).

Reply:
> Oh great! Should I help out with that one?

Well, it's a missing implementation so if I have the time I'll do it. In the mean, I let it as an open issue to anyone who wants to try it.